### ISSUE TITLE
Traces of the same tracing session from 2 different spans on the same shard may overwrite one another in a keyspace backend case

### ISSUE THREAD

**Issue by vladzcloudius** [scylladb/scylla#1681]
This is theoretical problem.
If there is a traced flow that that makes a loop and returns to the same shard, e.g.:

`Node1, shard1, span1` -> `Node2, shard2, span2` -> `Node1, shard1, span3`

Then, if we use a keyspace backend, traces created in `span1` and `span3` above may conflict if they are created in the same 100ns time slice - the one that is actually written last is going to win and the information from the one that would loose would be lost.

This is because of the primary key of a `system_traces.events` table is ((session ID), event ID), where `session ID` is a UUID and `event ID` is a time-UUID. 

In the example above `session ID` is going to be the same for all spans and if `event ID`s of traces in `span1` and `span3` are going to be the same they would effectively have the same primary key.

The proper solution for this problem would be to add the span ID to the `system_traces.events` primary key.

**Comment by slivne** [comment]
Changing the key may cause issues if there are tools that rely on this key - so I think we should try to avoid that if that is possible.

Is there another option instead of adding a field to the clustering key.

**Comment by slivne** [comment]
if we have a reproducer for this lets fix it -

**Comment by vladzcloudius** [comment]
@slivne No, I don't have a reproducer. This is a theoretical issue like I stated in the first line of the issue description.
